store/store.go

```diff
@@ -1255,9 +1257,69 @@ func (s *Store) downloadDelta(name string, downloadDir string, downloadInfo *sna
 		return "", err
 	}
 
+	// TODO: Check sha3_384
 	return deltaPath, nil
 }
 
+// applyDelta generates a target snap from the current snap and a downloaded delta.
+var applyDelta = func(name string, deltaPath string, deltaInfo *snap.DeltaInfo) (string, error) {
+	currentSnapName := fmt.Sprintf("%s_%d.snap", name, deltaInfo.FromRevision)
+	currentSnapPath := filepath.Join(dirs.SnapBlobDir, currentSnapName)
+
+	if !osutil.FileExists(currentSnapPath) {
+		return "", fmt.Errorf("the source snap at revision %d could not be found at %s", deltaInfo.FromRevision, currentSnapPath)
+	}
+
+	if deltaInfo.Format != "xdelta" {
+		return "", fmt.Errorf("unsupported delta format %q. Currently only \"xdelta\" format is supported", deltaInfo.Format)
+	}
+
+	targetSnapName := fmt.Sprintf("%s_%d_via_delta_%d_%d.snap", name, deltaInfo.ToRevision, deltaInfo.FromRevision, deltaInfo.ToRevision)
```

Comment:
This might read more easily, avoiding the repeated to revision:
```
("%s_%d_patched_from_%d.snap", name, deltaInfo.ToRevision, deltaInfo.FromRevision)